### PR TITLE
Marked files for translation

### PR DESCRIPTION
**Pull request by a-gradina**
Marked the following files for translation:

- `packages/desktop-client/src/components/reports/DateRange.tsx`
- `packages/desktop-client/src/components/reports/ReportOptions.ts`
- `packages/desktop-client/src/components/reports/disabledList.ts`
- `packages/desktop-client/src/components/reports/index.tsx`

### CODE REVIEW

File: packages/desktop-client/src/components/reports/DateRange.tsx [hunk 1]
```diff
@@ -15,7 +17,7 @@ type DateRangeProps = {
 
 function checkDate(date: string) {
   const dateParsed = new Date(date);
-  if (dateParsed.toString() !== 'Invalid Date') {
+  if (dateParsed.toString() !== t('Invalid Date')) {
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/DateRange.tsx:20]: **🔨 warning:** ‏this one should not be translated.

File: packages/desktop-client/src/components/reports/ReportOptions.ts [hunk 2]
```diff
@@ -16,11 +18,11 @@ export const defaultReport: CustomReportEntity = {
...
   endDate,
   isDateStatic: false,
-  dateRange: 'Last 6 months',
+  dateRange: t('Last 6 months'),
   mode: 'total',
   groupBy: 'Category',
-  interval: 'Monthly',
-  balanceType: 'Payment',
+  interval: t('Monthly'),
+  balanceType: t('Payment'),
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/ReportOptions.ts:25]: **❓ question:** ‏should these be translated? I'm not 100% sure. If they are saved in the DB, then I think they shouldn't be translated.

cc @carkom

File: packages/desktop-client/src/components/reports/ReportOptions.ts [hunk 3]
```diff
@@ -363,7 +365,7 @@ export const groupBySelections = (
       groupByLabel = 'category';
       break;
     default:
-      throw new Error('Error loading data into the spreadsheet.');
+      throw new Error(t('Error loading data into the spreadsheet.'));
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/ReportOptions.ts:368]: **💬 suggestion:** ‏exception messages should not be translated IMO
```suggestion
      throw new Error('Error loading data into the spreadsheet.');
```

File: packages/desktop-client/src/components/reports/disabledList.ts [hunk 4]
```diff
@@ -53,59 +55,59 @@ const totalGraphOptions: graphOptions[] = [
...
     disabledType: [],
-    defaultType: 'Payment',
+    defaultType: t('Payment'),
   },
   {
     description: 'AreaGraph',
-    disabledSplit: ['Category', 'Group', 'Payee', 'Account'],
-    defaultSplit: 'Interval',
+    disabledSplit: [t('Category'), t('Group'), t('Payee'), t('Account')],
+    defaultSplit: t('Interval'),
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/disabledList.ts:74]: **❓ question:** ‏should the strings in this file be translated?
cc @carkom

File: packages/desktop-client/src/components/reports/DateRange.tsx [hunk 5]
```diff
@@ -34,43 +36,59 @@ export function DateRange({ start, end, type }: DateRangeProps): ReactElement {
...
       </Text>
     );
   }
 
+  const formattedStartDate = d.format(startDate, 'MMM yyyy');
+  const formattedEndDate = d.format(endDate, 'MMM yyyy');
+  let typeOrFormattedEndDate: string;
+
+  if (type && ['budget', 'average'].includes(type)) {
+    typeOrFormattedEndDate = type;
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/DateRange.tsx:49]: **🔨 warning:** ‏
```suggestion
    typeOrFormattedEndDate = type === 'budget' ? 'budgeted' : type;
```

File: packages/desktop-client/src/components/reports/DateRange.tsx [hunk 6]
```diff
@@ -34,43 +36,59 @@ export function DateRange({ start, end, type }: DateRangeProps): ReactElement {
...
-        {type && 'Compare '}
-        {d.format(startDate, 'MMM yyyy')}
-        {type ? ' to ' : ' - '}
-        {['budget', 'average'].includes(type || '')
-          ? type
-          : d.format(endDate, 'MMM yyyy')}
+        {type ? (
+          <Trans
+            values={{
+              formattedStartDate,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/DateRange.tsx:77]: **💬 suggestion:** ‏could we use `startDate` for the sake of consistency?
```suggestion
              startDate: formattedStartDate,
```

File: packages/desktop-client/src/components/reports/DateRange.tsx [hunk 7]
```diff
@@ -34,43 +36,59 @@ export function DateRange({ start, end, type }: DateRangeProps): ReactElement {
...
+              endDate: typeOrFormattedEndDate,
+            }}
+          >
+            Compare {{ formattedStartDate }} to {{ endDate }}
+          </Trans>
+        ) : (
+          <Trans>
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/DateRange.tsx:84]: **❓ question:** ‏shouldn't this have arguments for the `Trans` component?

**Review by MatissJanis — CHANGES_REQUESTED** [review]

File: packages/desktop-client/src/components/reports/ReportOptions.ts (hunk 3 above)
Comment (a-gradina) [reply packages/desktop-client/src/components/reports/ReportOptions.ts:368]: I would've found it odd displaying them in other language but after googling it, it makes sense 😅
Thanks! I'll change it

File: packages/desktop-client/src/components/reports/disabledList.ts (hunk 4 above)
Comment (MatissJanis) [reply packages/desktop-client/src/components/reports/disabledList.ts:74]: I think these are saved in the DB too (but not 100% sure hence the tag of carkom). 

If they are saved in the DB - the values must remain in english, but we can change the labels shown to users (might require further code changes).

File: packages/desktop-client/src/components/reports/DateRange.tsx (hunk 7 above)
Comment (a-gradina) [reply packages/desktop-client/src/components/reports/DateRange.tsx:84]: Do you mean the value props? I just checked and they seem optional. I removed them, generate the JSON and it worked just as fine.

File: packages/desktop-client/src/components/reports/disabledList.ts (hunk 4 above)
Comment (a-gradina) [reply packages/desktop-client/src/components/reports/disabledList.ts:74]: Ah, I see. I'll wait with this then.

**Comment by MatissJanis** [comment]
While we wait for carkom to reply - would you mind taking a look at the VRT test? Something in this PR seems to be breaking it.

File: packages/desktop-client/src/components/reports/ReportOptions.ts (hunk 2 above)
Comment (carkom) [reply packages/desktop-client/src/components/reports/ReportOptions.ts:25]: They are saved in the DB in English. The app creates the report based on the values being in English.

File: packages/desktop-client/src/components/reports/disabledList.ts (hunk 4 above)
Comment (carkom) [reply packages/desktop-client/src/components/reports/disabledList.ts:74]: Same as above. They are saved to the DB in English and need to stay that way in order to properly re-create the saved report.

File: packages/desktop-client/src/components/reports/ReportOptions.ts (hunk 2 above)
Comment (a-gradina) [reply packages/desktop-client/src/components/reports/ReportOptions.ts:25]: > They are saved in the DB in English. The app creates the report based on the values being in English.

Alright, thanks. @MatissJanis I'll correct it in the evening.

**Comment by a-gradina** [comment]
> While we wait for carkom to reply - would you mind taking a look at the VRT test? Something in this PR seems to be breaking it.

Sure. I thought they were flaky, though. I had them failing in my previous PR as well a couple of times.

File: packages/desktop-client/src/components/reports/disabledList.ts [hunk 8]
```diff
@@ -1,41 +1,43 @@
+import { t } from 'i18next';
+
 const intervalOptions = [
   {
-    description: 'Daily',
-    defaultRange: 'This month',
+    description: t('Daily'),
+    defaultRange: t('This month'),
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/disabledList.ts:6]: cc @carkom should these be translated?

Comment (MatissJanis) [reply packages/desktop-client/src/components/reports/disabledList.ts:6]: bump @carkom

Comment (carkom) [reply packages/desktop-client/src/components/reports/disabledList.ts:6]: It's difficult to know. If you capture all of them then it should be fine. If it's translated then saved then all the options to reference are also translated then it should be fine as long as we capture them all.

However, if you save a report while the app is being translated and then switch back to English (or any other language) then the report will error out.

**Comment by carkom** [comment]
Sorry for the slow response all. I've been moving for most of last month. I added a comment to the conversation above. Not sure I'd recommend translating those bits without re-working how they are all saved.

File: packages/desktop-client/src/components/reports/disabledList.ts (hunk 8 above)
Comment (MatissJanis) [reply packages/desktop-client/src/components/reports/disabledList.ts:6]: Are any of these things stored in the DB? If tgey are then we definitely shouldn't be translating them.

Comment (MatissJanis) [reply packages/desktop-client/src/components/reports/disabledList.ts:6]: @a-gradina would you mind removing the translations from all the `defaultRange` values? I'm happy yo merge once we have that done :)

Sorry for this PR taking so long

Comment (MatissJanis) [reply packages/desktop-client/src/components/reports/disabledList.ts:6]: bump @a-gradina . I think there was a misunderstanding. You can keep the translations for `description`.  But remove from `defaultRange`.

**Review by MatissJanis — APPROVED** [review]
Thanks!

**Pull request merged**